recipes/cad1/task1/baseline/enhance.py

```diff
@@ -0,0 +1,309 @@
+""" Run the dummy enhancement. """
+import json
+import logging
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+import hydra
+import numpy as np
+import pandas as pd
+import torch
+from omegaconf import DictConfig
+from scipy.io import wavfile
+from torchaudio.pipelines import HDEMUCS_HIGH_MUSDB
+from torchaudio.transforms import Fade
+from tqdm import tqdm
+
+from clarity.enhancer.compressor import Compressor
+from clarity.enhancer.nalr import NALR
+
+logger = logging.getLogger(__name__)
+
+
+def separate_sources(
+    model,
+    mix,
+    sample_rate,
+    segment=10.0,
+    overlap=0.1,
+    device=None,
+):
+    """
+    Apply model to a given mixture. Use fade, and add segments together in order to add model segment by segment.
+
+    Arguments
+    ----------
+        model (torch.nn.Module): model to use for separation
+        mix (torch.Tensor): mixture to separate, shape (batch, channels, time)
+        sample_rate (int): sampling rate of the mixture
+        segment (int): segment length in seconds
+        overlap (float): overlap between segments, between 0 and 1
+        device (torch.device, str, or None): if provided, device on which to
+            execute the computation, otherwise `mix.device` is assumed.
+            When `device` is different from `mix.device`, only local computations will
+            be on `device`, while the entire tracks will be stored on `mix.device`.
+
+    Returns
+        torch.Tensor: estimated sources
+
+    Based on https://pytorch.org/audio/main/tutorials/hybrid_demucs_tutorial.html
+    """
+    if device is None:
+        device = mix.device
+    else:
+        device = torch.device(device)
+
+    mix = torch.as_tensor(mix, device=device)
+
+    if mix.ndim == 1:
+        # one track and mono audio
+        mix = mix.unsqueeze(0)
+    elif mix.ndim == 2:
+        # one track and stereo audio
+        mix = mix.unsqueeze(0)
+
+    batch, channels, length = mix.shape
+
+    chunk_len = int(sample_rate * segment * (1 + overlap))
+    start = 0
+    end = chunk_len
+    overlap_frames = overlap * sample_rate
+    fade = Fade(fade_in_len=0, fade_out_len=int(overlap_frames), fade_shape="linear")
+
+    final = torch.zeros(batch, len(model.sources), channels, length, device=device)
+
+    while start < length - overlap_frames:
+        chunk = mix[:, :, start:end]
+        with torch.no_grad():
+            out = model.forward(chunk)
+        out = fade(out)
+        final[:, :, :, start:end] += out
+        if start == 0:
+            fade.fade_in_len = int(overlap_frames)
+            start += int(chunk_len - overlap_frames)
+        else:
+            start += chunk_len
+        end += chunk_len
+        if end >= length:
+            fade.fade_out_len = 0
+
+    return final.cpu().detach().numpy()
+
+
+def get_device(device):
+    """Get device."""
+    if device is None:
+        if torch.cuda.is_available():
```

Comment:
`pylint` indicates the `else` is unnecessary ([no-else-return](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/no-else-return.html)).

Instead this can be written as...

```suggestion
        if torch.cuda.is_available():
            return torch.device("cude"), "cuda"
        return torch.device("cpu"), "cpu"
```